### PR TITLE
Internationalization Support

### PR DESCRIPTION
**Pull request by tsengia**
This PR will solve issue #297 by adding in internationalization support to Glowstone.

This is a duplicate of my other PR #315 because I committed to the wrong branch, which screws about everything else up. Sorry for the trouble :/

### CODE REVIEW

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 1]
```diff
@@ -324,14 +324,16 @@ private boolean migrate() {
         // server
         SERVER_IP("server.ip", "", Migrate.PROPS, "server-ip"),
         SERVER_PORT("server.port", 25565, Migrate.PROPS, "server-port"),
-        SERVER_NAME("server.name", "Glowstone Server", Migrate.PROPS, "server-name"),
+        SERVER_NAME("server.name", "Glowstone++ Server", Migrate.PROPS, "server-name"),
```
Comment (Paulomart) [inline src/main/java/net/glowstone/util/ServerConfig.java:327]: Without ++

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 2]
```diff
@@ -324,14 +324,16 @@ private boolean migrate() {
...
         LOG_FILE("server.log-file", "logs/log-%D.txt"),
         ONLINE_MODE("server.online-mode", true, Migrate.PROPS, "online-mode"),
         MAX_PLAYERS("server.max-players", 20, Migrate.PROPS, "max-players"),
         WHITELIST("server.whitelisted", false, Migrate.PROPS, "white-list"),
-        MOTD("server.motd", "Glowstone server", Migrate.PROPS, "motd"),
+        MOTD("server.motd", "Glowstone++ server", Migrate.PROPS, "motd"),
```
Comment (Paulomart) [inline src/main/java/net/glowstone/util/ServerConfig.java:332]: Without ++

File: src/main/java/net/glowstone/GlowServer.java [hunk 3]
```diff
@@ -358,38 +380,38 @@ private static ServerConfig parseArguments(String... args) {
...
-                System.out.println("  --host, -H <ip | hostname>     Sets the server listening address.");
-                System.out.println("  --onlinemode, -o <onlinemode>  Sets the server's online-mode.");
-                System.out.println("  --jline <true/false>           Enables or disables JLine console.");
-                System.out.println("  --plugins-dir, -P <directory>  Sets the plugin directory to use.");
-                System.out.println("  --worlds-dir, -W <directory>   Sets the world directory to use.");
-                System.out.println("  --update-dir, -U <directory>   Sets the plugin update folder to use.");
-                System.out.println("  --max-players, -M <director>   Sets the maximum amount of players.");
-                System.out.println("  --world-name, -N <name>        Sets the main world name.");
-                System.out.println("  --log-pattern, -L <pattern>    Sets the log file pattern (%D for date).");
+                logger.info("Available command-line options:");
```
Comment (VaiTon) [inline src/main/java/net/glowstone/GlowServer.java]: It's better use direct writing (System.out) for command line help argument

File: src/main/java/net/glowstone/GlowServer.java [hunk 4]
```diff
@@ -358,38 +380,38 @@ private static ServerConfig parseArguments(String... args) {
...
+                logger.info("  --update-dir, -U <directory>   Sets the plugin update folder to use.");
+                logger.info("  --max-players, -M <director>   Sets the maximum amount of players.");
+                logger.info("  --world-name, -N <name>        Sets the main world name.");
+                logger.info("  --log-pattern, -L <pattern>    Sets the log file pattern (%D for date).");
                 return null;
             } else if ("--version".equals(opt) || "-v".equals(opt)) {
-                System.out.println("Glowstone version: " + GlowServer.class.getPackage().getImplementationVersion());
-                System.out.println("Bukkit version:    " + GlowServer.class.getPackage().getSpecificationVersion());
-                System.out.println("Minecraft version: " + GAME_VERSION + " protocol " + PROTOCOL_VERSION);
+                logger.info("Glowstone version: " + GlowServer.class.getPackage().getImplementationVersion()); //glowstoneVersion
```
Comment (VaiTon) [inline src/main/java/net/glowstone/GlowServer.java]: It's better use direct writing (System.out) for command line version argument

**Comment by aramperes** [comment]
Suggestion: maybe we could use a I18N format like such instead:

``` ini
status.server.shutdown = The server is shutting down...
status.connection.ready = Ready for connections.
status.proxy.enabled = Proxy support is enabled.

warning.offlinemode = The server is running in offline mode! Only do this if you know what you're doing.
warning.option.invalid = Ignored invalid option:
warning.option.invalid.novalue = Ignored option specified without value: %s

error.bind.query = Failed to bind query. Address already in use?
error.bind.rcon = Failed to bind rcon. Address already in use?
error.bind.unknown = An unknown bind error has occurred.
error.reload = Uncaught error while reloading
error.plugin.load = Error loading %s
error.server.startup = Error during server startup.
error.proxy.parse = Failed to parse proxy data.
error.proxy.parse.for = Error parsing proxy data for %s
error.proxy.parse.invalid = Invalid proxy data provided.
error.handshake.client = Outdated client! I'm running %s
error.handshake.server = Outdated server! I'm running %s
error.state = Invalid state
error.command.internal = An internal error occurred while executing your command.

info.bind.success = Successfully bound to: %s
info.bind.address = Binding to address: %s

```

File: src/main/java/net/glowstone/GlowServer.java [hunk 5]
```diff
@@ -381,15 +403,15 @@ private static ServerConfig parseArguments(String... args) {
                 System.out.println("  --log-pattern, -L <pattern>    Sets the log file pattern (%D for date).");
                 return null;
             } else if ("--version".equals(opt) || "-v".equals(opt)) {
-                System.out.println("Glowstone version: " + GlowServer.class.getPackage().getImplementationVersion());
-                System.out.println("Bukkit version:    " + GlowServer.class.getPackage().getSpecificationVersion());
-                System.out.println("Minecraft version: " + GAME_VERSION + " protocol " + PROTOCOL_VERSION);
+                System.out.println("Glowstone++ version: " + GlowServer.class.getPackage().getImplementationVersion()); //glowstoneVersion
```
Comment (mastercoms) [inline src/main/java/net/glowstone/GlowServer.java:406]: Glowstone

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 6]
```diff
@@ -63,10 +63,10 @@ public ServerConfig(File configDir, File configFile, Map<Key, Object> parameters
         this.parameters = parameters;
 
         config.options().indent(4).copyHeader(true).header(
-                "glowstone.yml is the main configuration file for a Glowstone server\n" +
+                "glowstone.yml is the main configuration file for a Glowstone++ server\n" +
```
Comment (mastercoms) [inline src/main/java/net/glowstone/util/ServerConfig.java:66]: Glowstone

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 7]
```diff
@@ -63,10 +63,10 @@ public ServerConfig(File configDir, File configFile, Map<Key, Object> parameters
...
                         "It contains everything from server.properties and bukkit.yml in a\n" +
                         "normal CraftBukkit installation.\n\n" +
-                        "For help, join us on Gitter: https://gitter.im/GlowstoneMC/Glowstone");
+                        "For help, join us on Gitter: https://gitter.im/GlowstonePlusPlus/GlowstonePlusPlus");
```
Comment (mastercoms) [inline src/main/java/net/glowstone/util/ServerConfig.java:69]: Discord

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 8]
```diff
@@ -255,7 +255,7 @@ private void report(File file, InvalidConfigurationException e) {
             GlowServer.logger.log(Level.SEVERE, "Cannot load " + file + ": " + e.getCause().getClass(), e);
         }
     }
-
+
```
Comment (mastercoms) [inline src/main/java/net/glowstone/util/ServerConfig.java:258]: Extra tab

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 9]
```diff
@@ -377,7 +379,7 @@ private boolean migrate() {
         PERMISSIONS_FILE("files.permissions", "permissions.yml", Migrate.BUKKIT, "settings.permissions-file"),
         COMMANDS_FILE("files.commands", "commands.yml"),
         HELP_FILE("files.help", "help.yml"),
-
+
```
Comment (mastercoms) [inline src/main/java/net/glowstone/util/ServerConfig.java:382]: Extra tab

**Comment by mastercoms** [comment]
@ExpandingDev any updates on this?

**Comment by mastercoms** [comment]
Closing due to inactivity.

**Pull request closed without merging**